external/atomic_queue/include/atomic_queue/defs.h

```diff
@@ -31,6 +31,12 @@ static inline void spin_loop_pause() noexcept {
      defined(__ARM_ARCH_8A__) || \
      defined(__aarch64__))
     asm volatile ("yield" ::: "memory");
+#elif defined(__riscv)
```

Comment:
This is still within the `#elif defined(__arm__) || defined(__aarch64__)` above, so it will never actually apply. Also, the upstream atomic\_queue [already added RISC-V support](https://github.com/max0x7ba/atomic_queue/blob/083f3e5d666101e14008fc2084af4de4836c8bc5/include/atomic_queue/defs.h#L56) with something that looks potentially equivalent to what you've added here.

Reply:
Yes the upstream version is indeed the same thing as this one, despite that the .insn i pseudo-instruction is not supported until llvm >= 14 AFAIK

So basically if we use upstream's implementation then we'll be dropping support of some relatively older (but actually not that old) compilers. Which I think is okay because most RISC-V users know how to upgrade their compiler, but I'd like to know @redtide's opinion

Reply:
I think it's up to @paulfd to decide on this, IMO I would not care much about "old" compilers and use upstream to not to have to maintain patched stuff, so it would be nice to me if the upstream atomic_queue could be used as is and patch cpuid instead.

Reply:
Let's update to the latest cpuid. I'll make a note of it.

Reply:
#1249 but I wasn't precise: with "patch cpuid instead" I meant "patch cpuid __upstream__ instead", so @XieJiSS, could you please propose your cpuid changes there instead?

Reply:
Sure, I'll propose to them hopefully in the next few days

Reply:
Thank you very much!